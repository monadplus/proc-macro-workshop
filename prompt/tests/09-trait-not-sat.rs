use derive_prompt::*;

#[derive(Debug)]
struct B;

#[derive(Debug, FromPrompt)]
struct A {
    a1: u8,
    a2: B,
}

#[derive(Debug, FromPrompt)]
enum C {
    A(u8),
    B(B)
}

fn main() {}
