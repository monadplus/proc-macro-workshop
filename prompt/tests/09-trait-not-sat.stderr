error[E0277]: the trait bound `B: derive_prompt::Prompt` is not satisfied
 --> tests/09-trait-not-sat.rs:9:9
  |
9 |     a2: B,
  |         ^ the trait `derive_prompt::Prompt` is not implemented for `B`
  |
  = help: the following other types implement trait `derive_prompt::Prompt`:
            A
            C
            String
            Vec<T>
            bool
            char
            f32
            f64
          and $N others

error[E0277]: the trait bound `B: derive_prompt::Prompt` is not satisfied
  --> tests/09-trait-not-sat.rs:15:7
   |
15 |     B(B)
   |       ^ the trait `derive_prompt::Prompt` is not implemented for `B`
   |
   = help: the following other types implement trait `derive_prompt::Prompt`:
             A
             C
             String
             Vec<T>
             bool
             char
             f32
             f64
           and $N others
